### PR TITLE
FD2 packed binary commands

### PR DESCRIPTION
**Pull request by pgkeller**
### Ticket
#9012 

### Problem description
FD2 was not packing binaries

### What's changed
New prefetch/dispatch commands to handle large packed binaries

### Checklist
- [x] Post commit CI passes
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/impl/dispatch/kernels/cq_dispatch.cpp [hunk 1]
```diff
@@ -74,13 +74,20 @@ static uint32_t cmd_ptr;   // walks through pages in cb cmd by cmd
...
-    ((max_write_packed_cores + l1_to_local_cache_copy_chunk - 1) / l1_to_local_cache_copy_chunk) *
+
+// For packed write on GS, below must be>= 120 - 1. TODO: this should be a compile time arg passed in from host
+// For packed_write_large, must match CQ_DISPATCH_CMD_PACKED_WRITE_LARGE_MAX_SUB_CMDS
+constexpr uint32_t max_write_packed_cores = 108;
+constexpr uint32_t max_write_packed_large_cmd =
+    CQ_DISPATCH_CMD_PACKED_WRITE_LARGE_MAX_SUB_CMDS *
+    sizeof(CQDispatchWritePackedLargeSubCmd) / sizeof(uint32_t);
+constexpr uint32_t l1_cache_elements = (max_write_packed_cores > max_write_packed_large_cmd) ?
+    max_write_packed_cores : max_write_packed_large_cmd;
```
Comment (tt-asaigal) [inline tt_metal/impl/dispatch/kernels/cq_dispatch.cpp:85]: I don't fully understand whats happening here. Where is the value for `max_write_packed_cores ` coming from? This is what we have for a 1 row harvested Grayskull.
It seems like the `l1_cache_elements` are computed based on how many subcmds they need to store (copied using `careful_copy_from_l1_to_local_cache`)?

File: tt_metal/impl/dispatch/kernels/cq_prefetch.cpp [hunk 2]
```diff
@@ -249,7 +247,6 @@ void fetch_q_get_cmds(uint32_t& fence, uint32_t& cmd_ptr, uint32_t& pcie_read_pt
             // By here, prefetch_q_ready must be false
             // Nothing to fetch, nothing pending, nothing available, stall on host
             DEBUG_STATUS("HQW");
-            DPRINT << "fetch_q_get_cmds stall" << ENDL();
```
Comment (tt-asaigal) [inline tt_metal/impl/dispatch/kernels/cq_prefetch.cpp:252]: Some useful dprints that we're taking off here. Is this because of code binary size?

**Review by tt-asaigal — APPROVED** [review]

File: tt_metal/impl/dispatch/kernels/cq_prefetch.cpp (hunk 2 above)
Comment (pgkeller) [reply tt_metal/impl/dispatch/kernels/cq_prefetch.cpp:252]: sadly yes.  I considered wrapping all the dprints in a macro so we could enable them just when we want them

File: tt_metal/impl/dispatch/kernels/cq_dispatch.cpp (hunk 1 above)
Comment (pgkeller) [reply tt_metal/impl/dispatch/kernels/cq_dispatch.cpp:85]: there are 2 commands that use the buffer, the code above is calculating the size each one needs and picking the larger.  I could have done that manually and then just asserted that each type was <= to the manual value, that may be clearer, lmk

**Pull request closed without merging**